src/atlas/array/gridtools/GridToolsArrayView.cc

```diff
@@ -165,7 +176,11 @@ namespace array {
     template void ArrayView<long, Rank>::assign<true, nullptr>( std::initializer_list<long> const& );     \
     template void ArrayView<float, Rank>::assign<true, nullptr>( std::initializer_list<float> const& );   \
     template void ArrayView<double, Rank>::assign<true, nullptr>( std::initializer_list<double> const& ); \
-    template void ArrayView<long unsigned, Rank>::assign<true, nullptr>( std::initializer_list<long unsigned> const& );
+    template void ArrayView<long unsigned, Rank>::assign<true, nullptr>( std::initializer_list<long unsigned> const& ); \
+    template void ArrayView<int, Rank>::assign<true, nullptr>( ArrayView<int, Rank> const& );             \
```

Comment:
for understanding - why has assign got <true, nullptr> as tempate arguments?

Reply:
The assign() function has an extra (obscured) part of its signature by the definition
```
#define ENABLE_IF_NON_CONST                                                                               \
    template <bool EnableBool                                                                     = true, \
              typename std::enable_if<( !std::is_const<Value>::value && EnableBool ), int>::type* = nullptr>
```
This is made so that a "const" ArrayView will give compile errors, erasing the method.
```
    ENABLE_IF_NON_CONST
    void assign( const ArrayView<Value, Rank>& other );
```

Reply:
Aha! SINAE!

Reply:
... actually it is SFINAE in reverse!  Normally it is used to remove methods that cause compile failures. It is here being used to cause a compile failure!